service-providers/ip-packet-router/src/util/generate_new_ip.rs

```diff
@@ -1,38 +1,54 @@
-use std::{
-    collections::HashMap,
-    net::{IpAddr, Ipv4Addr},
-};
+use nym_ip_packet_requests::IPPair;
+use std::net::Ipv6Addr;
+use std::{collections::HashMap, net::Ipv4Addr};
 
-use crate::{constants::TUN_DEVICE_ADDRESS, mixnet_listener::ConnectedClient};
+use crate::constants::{TUN_DEVICE_ADDRESS_V4, TUN_DEVICE_ADDRESS_V6};
 
 // Find an available IP address in self.connected_clients
 // TODO: make this nicer
-fn generate_random_ip_within_subnet() -> Ipv4Addr {
+fn generate_random_ips_within_subnet() -> IPPair {
```

Comment:
is there any particular reason for having to use random ips rather than just using incremental values by keeping a counter?
i.e. first client would get `10.0.0.1`, the second client would get `10.0.0.2`, etc.

Reply:
I'm not sure why this is the approach @octol took, but we should talk and possibly change the approach here, as it can end up with inconsistent behaviour for users when close to reaching the limit, especially if we increase the size of the networks.

Reply:
I've seen other networking libs trying hard to avoid sequential client ids to avoid information leakage